test/Graphics/RenderStates.cpp

```diff
@@ -0,0 +1,80 @@
+#include <SFML/Graphics/RenderStates.hpp>
+#include "GraphicsUtil.hpp"
+
+#include <doctest.h>
+
+TEST_CASE("sf::RenderStates class - [graphics]")
+{
+    SUBCASE("Construction")
+    {
+        SUBCASE("Default constructor")
+        {
+            const sf::RenderStates renderStates;
+            CHECK(renderStates.blendMode == sf::BlendMode());
+            CHECK(renderStates.transform == sf::Transform());
+            CHECK(renderStates.texture == nullptr);
+            CHECK(renderStates.shader == nullptr);
+        }
+
+        SUBCASE("BlendMode constructor")
+        {
+            const sf::BlendMode blendMode(sf::BlendMode::Zero, sf::BlendMode::SrcColor, sf::BlendMode::ReverseSubtract,
+                                          sf::BlendMode::OneMinusDstAlpha, sf::BlendMode::DstAlpha, sf::BlendMode::Max);
+            const sf::RenderStates renderStates(blendMode);
+            CHECK(renderStates.blendMode == blendMode);
+            CHECK(renderStates.transform == sf::Transform());
+            CHECK(renderStates.texture == nullptr);
+            CHECK(renderStates.shader == nullptr);
+        }
+
+        SUBCASE("Transform constructor")
+        {
+            const sf::Transform transform(10, 9, 8, 7, 6, 5, 4, 3, 2);
+            const sf::RenderStates renderStates(transform);
+            CHECK(renderStates.blendMode == sf::BlendMode());
+            CHECK(renderStates.transform == transform);
+            CHECK(renderStates.texture == nullptr);
+            CHECK(renderStates.shader == nullptr);
+        }
+
+        SUBCASE("Texture constructor")
+        {
+            const sf::Texture* texture = nullptr;
```

Comment:
You can't create an actual instance of a Texture because of the headless setup?

Reply:
That's right. it's probably worth us figuring out how to get a "headed" Linux CI runner since it's going to cause issues with later unit tests. I tried writing `sf::Drawable` tests last night and had to construct an instance of `sf::RenderWindow` but it had the same issue with the Linux jobs failing so that class just cannot be tested right now.